src/sds.c

```diff
@@ -254,8 +254,10 @@ sds _sdsMakeRoomFor(sds s, size_t addlen, int greedy) {
     if (greedy == 1) {
         if (newlen < SDS_MAX_PREALLOC)
             newlen *= 2;
```

Comment:
You may have missed it here, if `newlen` is already overflow, `hdrlen + newlen + 1 > newlen` will pass.

Reply:
> You may have missed it here, if `newlen` is already overflow, `hdrlen + newlen + 1 > newlen` will pass.
Since SDS_MAX_PREALLOC is defined as 1024*1024, so it can not overflow here.

Reply:
Ohh, You are right. Let me look again.